src/constructs/s3/index.ts

```diff
@@ -19,3 +22,130 @@ export class GuS3Bucket extends GuAppAwareConstruct(Bucket) {
     });
   }
 }
+
+interface GuS3OriginBucketDeploymentProps {
```

Comment:
Only a subset of the available [parameters](https://riffraff.gutools.co.uk/docs/magenta-lib/types#awss3) are configurable, on the basis that the rest have sensible defaults and only impact the directory structure of the bucket's content.